### PR TITLE
Refactor CLI

### LINKED ISSUE #52: Add default project config and enable CLI configuration
This issue is related to the **Project Usability** item in the project [overview](https://hackmd.io/@brech1/circom-to-arithc-overview).

To improve usability we need to create our own CLI argument parsing and remove the existent one in the `circom` directory. This one has been built around regular circom usage and needs to be removed, along with any configuration we're not using or planning to.

A project configuration struct has to be made, starting from:

```rust
struct CompilerConfig {
    /// Circuit input file.
    input: String,
    /// Circuit and report output paths.
    output: String,
}
```

This will be helpful when using our project as an executable from a command line interface, but can also serve as inspiration if the project is used as a library.

This will also allow us to update the `clap` dependency.

## Tasks

- [ ] Removal of all code in the `circom` directory related to CLI arguments parsing or project configuration.
- [ ] Implementation of our own CLI, with a `compile` command. 
- [ ] Implementation of a new `CompilerConfig` struct that holds all necessary optional data related to project compilation. Keep in mind this would only be related to the `compile` command on the CLI.
- [ ] Update `clap` dependency.

### PR DESCRIPTION
**Pull request by curryrasul**
## Description
The PR aimes to remove unnecessary circom's code for CLI; update it for the current needs with a newer clap crate version

## Issue
This PR aimed to close #52

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -4,6 +4,17 @@
...
+pub mod cli;
 pub mod process;
 pub mod program;
 pub mod runtime;
+
+use std::path::{Path, PathBuf};
+
+pub use cli::Args;
+
+pub fn build_output(output_path: &Path, filename: &str, ext: &str) -> PathBuf {
```
Comment (brech1) [inline src/lib.rs:16]: Perhaps this function could be included in some module? As to keep `lib.rs` to only deal with exports

Comment (curryrasul) [reply src/lib.rs:16]: I can create utils.rs and put this function there, no probs:)

In general it's not bad to have functions in lib.rs as well

Comment (brech1) [reply src/lib.rs:16]: That's ok! I would keep it here rather than creating a new one 😅

**Review by brech1 — APPROVED** [review]
LGTM!

**Comment by curryrasul** [comment]
@voltrevo Could you review this as well please? Because I was resolving merge conflict with your changes on main. I might have missed something ...

**Pull request merged**